### PR TITLE
add parameters to maven profile

### PR DESCRIPTION
**Pull request by savroman**

### CODE REVIEW

**Comment by bdeputat** [comment]
Hello Roman! I have some questions:
1. Why you declare a variable: 
$archive = "apache-maven-${version}-bin.tar.gz"
And then when you declare the next variable 
$source = "${maven_url}${version}/binaries/apache-maven-${version}-bin.tar.gz",
you do not use: $archive?
Maybe it's better to announce it this way:
$source = "${maven_url}${version}/binaries/$archive"

2. It is not clear why you are in the profile: /profile/manifests/maven.pp,
declare the parameters:
+  $version = "3.3.0",
+  $path    = "/usr/",
+  ) {
+  class maven3 {
+    maven_version => $version,
+    install_path  => $path,
but in the manifesto: /maven3/manifests/init.pp 
declare the same parameters as:
+  $maven_version = '3.5.3',
+  $load_dir = "/tmp/",
+  $maven_url = "http://archive.apache.org/dist/maven/maven-3/",
+  $install_path = "/opt/",

Hope the comments will be useful to you.

File: modules/maven3/manifests/init.pp [hunk 1]
```diff
@@ -5,37 +5,45 @@
...
 #  include maven3
 #
-#  class { 'maven3': 
+#  class { 'maven3':
 #    version => '3.5.3',
 #  }
 
 class maven3 (
-  $version = '3.5.3',
+  $maven_version = '3.5.3',
```
Comment (yurkovskiy) [inline modules/maven3/manifests/init.pp]: Why version number was provided in code? As I remember the deal was to using parameters for providing version numbers

Comment (savroman) [reply modules/maven3/manifests/init.pp]: I used it as a default value. You can overwrite it when class is called.

Comment (ipleten) [reply modules/maven3/manifests/init.pp]: lets stick to "latest"  or "installed" as default value for version of any package.

File: modules/maven3/manifests/init.pp [hunk 2]
```diff
@@ -5,37 +5,45 @@
...
+  $load_dir = "/tmp/",
+  $maven_url = "http://archive.apache.org/dist/maven/maven-3/",
+  $install_path = "/opt/",
   ){
 
   $archive = "apache-maven-${version}-bin.tar.gz"
-  $source = "http://archive.apache.org/dist/maven/maven-3/${version}/binaries/apache-maven-${version}-bin.tar.gz"
-  $load_dir = "/tmp/"
+  $source = "${maven_url}${version}/binaries/apache-maven-${version}-bin.tar.gz"
```
Comment (ipleten) [inline modules/maven3/manifests/init.pp]: as @bdeputat  has mentioned: reuse of $archive is a good idea. I'd rename it to $archive_name.

File: modules/maven3/manifests/init.pp [hunk 3]
```diff
@@ -5,37 +5,45 @@
...
-#  class { 'maven3': 
+#  class { 'maven3':
 #    version => '3.5.3',
 #  }
 
 class maven3 (
-  $version = '3.5.3',
+  $maven_version = '3.5.3',
+  $load_dir = "/tmp/",
+  $maven_url = "http://archive.apache.org/dist/maven/maven-3/",
```
Comment (ipleten) [inline modules/maven3/manifests/init.pp]: Where is our REPO?!

File: modules/maven3/manifests/init.pp [hunk 4]
```diff
@@ -5,37 +5,45 @@
...
+  $source = "${maven_url}${version}/binaries/apache-maven-${version}-bin.tar.gz"
   $archive_path = "${load_dir}${archive}"
 
-  exec { 'upload_maven_archive':
-    command => "wget ${source}",
-    path    => '/bin:/sbin:/usr/bin:/usr/sbin:',
-    cwd     => "${load_dir}",
-    creates => "${archive_path}",
-    before  => Exec['maven-untar'],
+  if !("jdk1" in $facts['path']) {
```
Comment (ipleten) [inline modules/maven3/manifests/init.pp]: I'm not sure that we rally need that check

File: modules/profile/manifests/maven.pp [hunk 5]
```diff
@@ -1,4 +1,9 @@
-class profile::maven {
-  include java8
-  include maven3
+class profile::maven (
+  $version = "3.3.0",
```
Comment (ipleten) [inline modules/profile/manifests/maven.pp]: params in profile :(

**Review by ipleten — CHANGES_REQUESTED** [review]

File: modules/maven3/manifests/init.pp [hunk 6]
```diff
@@ -5,37 +5,44 @@
...
-#  class { 'maven3': 
+#  class { 'maven3':
 #    version => '3.5.3',
 #  }
 
 class maven3 (
   $version = '3.5.3',
+  $load_dir = "/tmp/",
+  $source = "http://archive.apache.org/dist/maven/maven-3/${version}/binaries/",
+  $archive = "apache-maven-${version}-bin.tar.gz",
```
Comment (ipleten) [inline modules/maven3/manifests/init.pp]: as @bdeputat  has suggested using $archive is a good idea.

File: modules/maven3/manifests/init.pp [hunk 7]
```diff
@@ -5,37 +5,44 @@
...
   $archive_path = "${load_dir}${archive}"
 
-  exec { 'upload_maven_archive':
-    command => "wget ${source}",
-    path    => '/bin:/sbin:/usr/bin:/usr/sbin:',
-    cwd     => "${load_dir}",
-    creates => "${archive_path}",
-    before  => Exec['maven-untar'],
+  if !("jdk" in $facts['path']) {
+    warning('No JDK installed')
```
Comment (ipleten) [inline modules/maven3/manifests/init.pp]: I have a deja vu.

That check is useless i think.

File: modules/maven3/manifests/init.pp [hunk 8]
```diff
@@ -5,37 +5,44 @@
...
 #
-#  class { 'maven3': 
+#  class { 'maven3':
 #    version => '3.5.3',
 #  }
 
 class maven3 (
   $version = '3.5.3',
+  $load_dir = "/tmp/",
+  $source = "http://archive.apache.org/dist/maven/maven-3/${version}/binaries/",
```
Comment (ipleten) [inline modules/maven3/manifests/init.pp:15]: Where is our repo with mvn.
Stop wget and curl jars tar zips from the Internet use RPM.

**Review by ipleten — CHANGES_REQUESTED** [review]

File: modules/maven3/manifests/init.pp (hunk 2 above)
Comment (savroman) [reply modules/maven3/manifests/init.pp]: accepted

File: modules/profile/manifests/maven.pp (hunk 5 above)
Comment (savroman) [reply modules/profile/manifests/maven.pp]: THX. Forget to correct. Will be moved as a class in jenkins profile/ I think we don't need separated maven profile/

File: modules/maven3/manifests/init.pp (hunk 7 above)
Comment (savroman) [reply modules/maven3/manifests/init.pp]: accepted

File: modules/maven3/manifests/init.pp (hunk 8 above)
Comment (savroman) [reply modules/maven3/manifests/init.pp:15]: I'm going to use it until repo will be finished. Will be fixed in future

**Review by oleksdiam — APPROVED** [review]
Everything I've seen is acceptable in my opinion

**Review by ipleten — APPROVED** [review]

**Pull request merged**